python/ray/train/v2/_internal/execution/worker_group/worker_group.py

```diff
@@ -229,19 +166,52 @@ def start(
             WorkerGroupStartupFailedError: If the worker group fails to start
                 due to actors dying/failing during initialization.
         """
+        assert not self.has_started(), "Worker group already started."
+
+        worker_group_state_builder = WorkerGroupStateBuilder()
+
+        try:
+            self._start(
+                worker_group_state_builder,
+                train_fn,
+                num_workers,
+                resources_per_worker,
+                placement_strategy,
+                checkpoint,
+            )
+        except Exception as e:
+            if not self.has_started():
+                worker_group_state_builder.shutdown()
+            raise e
```

Comment:
Why conditional on if it has not started? Is it even possible for `has_started` to be True here?

Reply:
My train of thought for this was that once `has_started` is True, then there is a `WorkerGroupState` which should be acted on, and the `Builder` should basically be discarded and not operated on anymore.

Right now in the code it should not return True here. This could also be changed to `assert not self.has_started()`.